### PR TITLE
Fix broken e2e test

### PR DESCRIPTION
**Pull request by dg0yt**
Noticed while trying to implement a new test.

### CODE REVIEW

**Comment by dg0yt** [comment]
The test is fixed. Can you assign someone to fix vcpkg-tool?

File: azure-pipelines/end-to-end-tests-dir/build-test-ports.ps1 [hunk 1]
```diff
@@ -22,12 +22,12 @@ if ($IsWindows) {
 
 $output = Run-VcpkgAndCaptureOutput @commonArgs --overlay-ports="$PSScriptRoot/../e2e_ports/overlays" install broken-no-name
 Throw-IfNotFailed
-if (-not $output -match 'missing field') {
+if (-not ($output -match 'missing field')) {
```
Comment (ras0219-msft) [inline azure-pipelines/end-to-end-tests-dir/build-test-ports.ps1:25]:
```suggestion
if (-not ($output -match "missing required field 'name'")) {
```

File: azure-pipelines/end-to-end-tests-dir/build-test-ports.ps1 [hunk 2]
```diff
@@ -22,12 +22,12 @@ if ($IsWindows) {
...
 Throw-IfNotFailed
-if (-not $output -match 'missing field') {
+if (-not ($output -match 'missing field')) {
     throw 'Did not detect missing field'
 }
 
 $output = Run-VcpkgAndCaptureOutput @commonArgs --overlay-ports="$PSScriptRoot/../e2e_ports/overlays" install broken-no-version
 Throw-IfNotFailed
-if (-not $output -match 'missing field') {
+if (-not ($output -match 'missing field')) {
```
Comment (ras0219-msft) [inline azure-pipelines/end-to-end-tests-dir/build-test-ports.ps1:31]:
```suggestion
if (-not ($output -match 'expected a versioning field')) {
```

**Comment by dg0yt** [comment]
All done. All green again.

**Comment by BillyONeal** [comment]
Stupid question: Why is it broken but passing in PRBuild?

**Comment by dg0yt** [comment]
> Stupid question: Why is it broken but passing in PRBuild?

Before this PR, it was *always* passing, regardless of compared strings. The test was ineffective.
The first commit changed this to actually fail when the actual string didn't match the expexted string.
Based on feedback from @ras0219-msft, the second commit updated the expected string, making the test pass again for the current source code. (If the test hadn't been broken, it would have been changed together with tool source code.)

No further fixes needed.

**Comment by BillyONeal** [comment]
>Before this PR, it was always passing, regardless of compared strings. The test was ineffective.

That explains it. Thanks!

**Pull request merged**